### ISSUE TITLE
Pulse slackbot needs TLC

### ISSUE THREAD

**Issue by xmatthewx** [MozillaFoundation/mofo-devops#555]
The creator field needs a tweak according to @alanmoo, b/c of an update. While we're at it, should move update the mofo slack channel to use our bot instead of Phillips for maintenance.

**Comment by alanmoo** [comment]
Instead of an array of creators as strings, it’s now `related_creators` which is an array of objects with `name` properties. For an example, check out https://api.mozillapulse.org/api/pulse/entries/2/

**Comment by cadecairos** [comment]
@alanmoo we're going to need a paid account on Zapier to keep running our own bot. once our free trial ran out they disabled the bot we set up for MozFest.

**Comment by alanmoo** [comment]
Hmmm. I don't know that this one thing is worth paying for a Zapier account. Personally I don't check the slack channel much- I have new posts in there muted to reduce the overwhelming firehose. @xmatthewx?

**Comment by xmatthewx** [comment]
Could we accomplish this with IFTTT? There's a large percentage of staff in the moco slack channel, more still arriving, and people mention to me they follow it. Seems useful for staff awareness. What's the cost?

**Comment by alanmoo** [comment]
Not sure IFTTT is as advanced as we need. If people are using it, yeah, it might make sense to pay for, I just wasn't sure.

**Comment by xmatthewx** [comment]
I'll add this to our next weekly with Jesse.

**Comment by cadecairos** [comment]
labelling this as blocked, on the meeting with Jesse.

**Comment by xmatthewx** [comment]
And.... unblocked. $20 a month is well worth it. 👍 from our friend with the money bags.

**Comment by cadecairos** [comment]
Update: got our subscription. We're working on updating the Zap to support the `related_creators` field.

**Comment by xmatthewx** [comment]
I know this stuff isn't super high priority, but it would be great to have it up before all hands so we can get activity rolling again.

**Comment by patjouk** [comment]
I worked on it all yesterday (it was really frustrating because of zapier): it should work now but I want to do more testing today. I'll keep you posted :)

**Comment by patjouk** [comment]
And it's back!
To be honest, the code is probably a bit fragile (thanks zapier ><) so the next steps should be to add RSS support and to use it to generate the slack message.

**Comment by xmatthewx** [comment]
@patjouk @cadecairos - the bot hasn't been working for while. Should we file a new ticket or reopen this one?

**Comment by cadecairos** [comment]
We had to disable it because breaking changes to the pulse API keep breaking the integration. 

We'd like to either get an RSS feed endpoint set up or have the endpoints properly versioned before turning it on again.

**Comment by alanmoo** [comment]
That's more motivation to version the pulse API as part of the fellowships work.

**Comment by xmatthewx** [comment]
Whether we solve this longterm with versioning or setting up RSS, let's keep it as a high priority. It's pretty essential for staff engagment ... and moving to twitter ... engagement with fellows and a broader audience. 

In the mean time, what if we grab less info: just image, title, url. Cut out the rest.

**Comment by patjouk** [comment]
Hey @xmatthewx. The previous bot was using creator fields [that got deprecated](https://github.com/mozilla/network-pulse-api/pull/252/commits/e81d15a7902acad8e4978527674b3d1c9612a246). I [wrote a prototype](https://github.com/MozillaFoundation/mofo-devops/blob/master/slackbots/latest_pulse.md) as a quick fix for Mozfest slack. It was okay-ish during the festival but kept on breaking after that :(. Sadly, I can't do another quick fix that removes stuff. I would prefer to find a more resilient and long-term solution for that bot, especially if it needs to be used on Twitter.

Now that the work on the fake data for the foundation site [is nearly done](https://github.com/mozilla/foundation.mozilla.org/pull/932), I'll go back to pulse and add some RSS. That should fix the bot 🤞

**Comment by xmatthewx** [comment]
thanks @patjouk! rss would be great!

**Comment by cadecairos** [comment]
This is resolved!
